### ISSUE TITLE
Improve accessibility for Pagination component

### ISSUE THREAD

**Issue by ilyas-sov** [carbon-design-system/carbon#19316]
### The problem

Currently the `Pagination` component from the `"@carbon/react"` library is not focusable by default. Adding `tabindex={0}` makes it focusable but breaks the accessibility tests because of absence of proper `role` attribute. Adding role `navigation` or `group` doesn't help because violates SonarQube rules which suggests to use `<nav>` or `<fieldset>` or others semantic html tags instead.

### The solution

Add `as` prop for customizing the pagination container html or use semantic `<nav>` tag.

### Package

@carbon/react

### Business priority

Medium Priority = upcoming release but is not pressing

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by preetibansalui** [comment]
Hi @ilyas-sov, The pagination component's keyboard navigation appears to be working correctly according to the [documentation](https://carbondesignsystem.com/components/pagination/accessibility/). Could you please share more details about your use case and why you need the entire component to receive focus initially?

**Comment by ilyas-sov** [comment]
Hi @preetibansalui, thank you for quick response.
I had a business requirement for such functionality for easier navigation, but you are right, according to documentation navigation works correctly. I discussed it again with my team and I will close this issue. Sorry for confusing.
